src/workerd/api/node/tests/http-server-nodejs-test.js

```diff
@@ -0,0 +1,636 @@
+import http from 'node:http';
+import { strictEqual, ok, throws, notStrictEqual, rejects } from 'node:assert';
+import { nodeCompatHttpServerHandler } from 'cloudflare:workers';
+import { mock } from 'node:test';
+
+export const checkPortsSetCorrectly = {
+  test(_ctrl, env) {
+    const keys = ['PONG_SERVER_PORT'];
+    for (const key of keys) {
+      strictEqual(typeof env[key], 'string');
+      ok(env[key].length > 0);
+    }
+  },
+};
+
+const globalServer = http.createServer();
+globalServer.listen(9090);
+
+export const testGlobalHttpServe = {
+  async test(_ctrl, env) {
+    strictEqual(globalServer.listening, true);
+    strictEqual(globalServer.address().port, 9090);
+    globalServer.close();
+  },
+};
+
+// Test is taken from test/parallel/test-http-server-async-dispose.js
+export const testHttpServerAsyncDispose = {
+  async test() {
+    const server = http.createServer();
+
+    server.listen(8080);
+    ok(server.listening);
+    const closeFn = mock.fn();
+    server.on('close', closeFn);
+    await server[Symbol.asyncDispose]();
+    ok(!server.listening);
+    strictEqual(closeFn.mock.callCount(), 1);
+  },
+};
+
+// Test is taken from test/parallel/test-http-server-incomingmessage-destroy.js
+export const testHttpServerIncomingMessageDestroy = {
+  async test(_ctrl, env) {
+    await using server = http.createServer((req, res) => {
+      req.on('error', (err) => {
+        res.statusCode = 400;
+        res.end('Request destroyed: ' + err.message);
+      });
+      req.destroy(new Error('Destroy test'));
```

Comment:
There should also be an example where `req.destroy()` is called without an error as that would mean no `'error'` event is emitted. Also, Is there a test that verifies what happens if `res.destroy()` and `res.destroy(error)` are called?